### PR TITLE
Object.prototype.toString instead of instanceOf to determine arrayness

### PR DESCRIPTION
**Pull request by MonkeyDZeke**
Resolves https://github.com/mholt/PapaParse/issues/235

### CODE REVIEW

**Comment by MonkeyDZeke** [comment]
I'm puzzled by that as well. I had thought it might be something to do with my local build since that error shows up on master as well for me. Evidently not. :thinking:

**Comment by pokoli** [comment]
It seems that something changed and our test suite is broken now (all PR are failing).

I think we should fix our test suite before merging any new code.

it seems a problem with the chai.js version [which is loaded from their website ](https://github.com/mholt/PapaParse/blob/master/tests/tests.html#L9). Probably we should include or own version of chai.js with a pinned dependency

**Comment by pokoli** [comment]
Regarding the PR could you please add a test suite that test the isArray function against a basic case and some failing case?

**Comment by MonkeyDZeke** [comment]
I could I suppose, but it's literally a direct copy of the [official polyfill](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray#Polyfill) for `Array.isArray`. And it's hard to test the original failing case because it only fouls up when the array is set in a different frame from where it's tested (from what I understand).

**Comment by pokoli** [comment]
@MonkeyDZeke Then I think we should use the official polyfill when available, and define all needed polyfils on the top of the papaparse.js file.

**Comment by MonkeyDZeke** [comment]
@pokoli Like so?

File: papaparse.js [hunk 1]
```diff
@@ -41,6 +41,14 @@
 		return {};
 	})();
 
+	// Polyfills
```
Comment (pokoli) [inline papaparse.js]: I think we should also define in the parent scope so

**Comment by MonkeyDZeke** [comment]
@pokoli Is that what you were meaning?

File: papaparse.js [hunk 2]
```diff
@@ -1,9 +1,19 @@
 /*@license
-	Papa Parse
-	v4.6.0
-	https://github.com/mholt/PapaParse
-	License: MIT
+Papa Parse
+v4.6.0
+https://github.com/mholt/PapaParse
```
Comment (pokoli) [inline papaparse.js]: why changing license?

**Comment by pokoli** [comment]
@MonkeyDZeke yes, that's what I was looking for

**Comment by MonkeyDZeke** [comment]
Cool, should be good then, once we figure out what's breaking the test suite.

**Comment by pokoli** [comment]
Could you rebase with latest mater branch so we could ensure the test is passing and then merge it? 

Thanks

**Comment by MonkeyDZeke** [comment]
Rebased and ready for merging. :smile:

**Pull request merged**